mergebot.py

```diff
@@ -1,195 +1,54 @@
-  if not checkout_success == 0:
-    post_error('Error checking out target branch: master. Please try again.', pr)
-    return False
-  print("Checked out Apache master.")
-
-  # Merge
-  merge_success = call(['git', 'merge', '--no-ff', '-m', 'This closes #{}'.format(pr), 'finish-pr-{}'.format(pr)], cwd='/usr/local/google/home/jasonkuster/tmp/')
-  if not merge_success == 0:
-    post_error('Merge was not successful against target branch: master. Please try again.', pr)
-    return False
-  print("Merged successfully.")
-
-  print("Running mvn clean verify.")
-  # mvn clean verify
-  mvn_success = call(['mvn', 'clean', 'verify'], cwd='/usr/local/google/home/jasonkuster/tmp/')
-  if not mvn_success == 0:
-    post_error('mvn clean verify against HEAD + PR#{} failed. Not merging.'.format(pr), pr)
-    return False
-
-  # git push (COMMENTED UNTIL MERGEBOT HAS PERMISSIONS)
-  #push_success = call(['git', 'push', 'apache', 'HEAD:master'], cwd='/usr/local/google/home/jasonkuster/tmp/')
-  #if not push_success == 0:
-  #  post_error('Git push failed. Please try again.', pr)
-  #  return False
-  return True
-
-
-def post_error(content, pr_num):
-  post_pr_comment("Error: {}, #{}.".format(content, pr_num))
-
-
-def post_info(content, pr_num):
-  post_pr_comment("Info: {}, #{}.".format(content, pr_num))
-
-
-def post_pr_comment(content, pr_num):
-  print(content)
-  post(content, COMMENT_FMT_URL.format(pr_num=pr_num))
-
-
-def post(content, endpoint):
-  payload = {"body": content}
-  requests.post(endpoint, data=payload)
-
-
-def set_up():
-  if not call(['mkdir', '/usr/local/google/home/jasonkuster/tmp']) == 0:
-    return False
-  return True
-
-
-def clean_up():
-  if not call(['rm', '-rf', '/usr/local/google/home/jasonkuster/tmp']) == 0:
-    return False
-  return True
-
-
-if __name__ == "__main__":
+  print('Mergebot Manager Starting Up')
+
+  configs = []
+  print('Parsing Config Files')
+  for fn in glob.iglob('config/*.yaml'):
+    with open(fn) as cfg:
+      try:
+        print('Opening {}'.format(fn))
+        config = yaml.load(cfg)
+        print('{} Successfully Read'.format(fn))
+        configs.append(config)
+      except YAMLError as exc:
+        print("Error parsing file {}: {}. Continuing with others.".format(fn, exc))
+
+  # Workaround for multiprocessing SIGINT problems
+  original_sigint_handler = signal.signal(signal.SIGINT, signal.SIG_IGN)
+  pool = multiprocessing.Pool(len(configs))
+  signal.signal(signal.SIGINT, original_sigint_handler)
+
+  try:
+    print('Starting Poller Pool')
+    pool.map_async(poll_scm, configs)
+    while True:
+      input("Mergebot running; press Ctrl + C to cancel.\n")
+  except KeyboardInterrupt:
+    print("\nExiting.")
+    pool.terminate()
+  else:
+    # Generally this shouldn't happen - pollers should run forever.
+    pool.close()
+    pool.join()
+    print('Mergebot terminated early - all pollers must have crashed.')
```

Comment:
When a poller crashes is it not restarted? Then we should make sure to clearly indicate when a poller dies so that we don't get into a state where this is running, but only polling one repository.